### ISSUE TITLE
`!FindInMap` issues while in `Fn::ForEach`, cannot find resource or parameter 

### ISSUE THREAD

**Issue by jpSimkins** [aws-cloudformation/cfn-lint#3198]
### CloudFormation Lint Version

cfn-lint 0.87.0

### What operating system are you using?

Ubuntu

### Describe the bug

I am now getting:
> E1012 Ref AccountId not found as a resource or parameter

When using `!FindInMap` in a `Fn::ForEach`

### Expected behavior

I expected this to pass as this deploys fine.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::LanguageExtensions'

Description: Configures IAM Identity Center. IAM Identity Center must be enabled before you create this stack.

Parameters:

  IamIdentityCenterInstanceArn:
    Type: String
    Description: 'The ARN for the IAM Identity Center Instance. Found in the settings of IAM Identity Center.'

  IdentityStoreId:
    Type: String
    Description: 'The id for the identity store. Found in the settings of IAM Identity Center.'


Mappings:

  AccountIdsMap:
    # GovCloud
    "101010101010":
      DevSandAccount: "1111111111111"
      ProjectADevAccount: "22222222222222"
      ProejctADemoAccount: "333333333333333"
    # Commercial
    "101010101011":
      DevSandAccount: "1111111111112"
      ProjectADevAccount: "22222222222223"
      ProejctADemoAccount: "333333333333334"


Resources:

  SystemAdminPermissionSet:
    Type: AWS::SSO::PermissionSet
    Properties:
      InstanceArn: !Ref IamIdentityCenterInstanceArn
      Name: 'SystemAdmin'
      Description: 'System Admin Permissions'
      SessionDuration: 'PT8H'
      InlinePolicy:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Action: '*'
            Resource: '*'

  SystemAdminGroup:
    Type: AWS::IdentityStore::Group
    Properties:
      DisplayName: SystemAdmin
      Description: 'System Admin Group'
      IdentityStoreId: !Ref IdentityStoreId

          
  # Testing Account Assignments
  'Fn::ForEach::SystemAdminAssignments':
    - AccountId
    - - DevSandAccount
      - ProjectADevAccount
      - ProejctADemoAccount
    - 'SystemAdminAssignment${AccountId}':
        Type: 'AWS::SSO::Assignment'
        Properties:
          InstanceArn: !Ref IamIdentityCenterInstanceArn
          PermissionSetArn: !GetAtt SystemAdminPermissionSet.PermissionSetArn
          TargetId: !FindInMap
            - AccountIdsMap
            - !Ref AWS::AccountId
            - !Ref AccountId
          TargetType: AWS_ACCOUNT
          PrincipalId: !GetAtt SystemAdminGroup.GroupId
          PrincipalType: GROUP
```

**Comment by kddejong** [comment]
Thanks for reporting this.  Built in some logic to clear each of the elements in a FindInMap when have issues resolving it.  Should have this released today.  I'm waiting on the new specs to be released today then I'll do a release.
